frontend/src/Components/cards/HelpfulLinkCard.tsx

```diff
@@ -38,38 +38,29 @@ export default function HelpfulLinkCard({
     function changeVisibility(visibilityStatus: boolean) {
         if (visibilityStatus == !visible) {
             setVisible(visibilityStatus);
-            void handleToggleVisibility();
+            void handleToggleAction('toggle');
         }
     }
 
-    const handleToggleVisibility = async () => {
+    const handleToggleAction = async (
+        action: 'favorite' | 'toggle',
+        e?: React.MouseEvent
+    ) => {
+        if (e) e.stopPropagation();
         const response = await API.put<null, object>(
-            `helpful-links/toggle/${link.id}`,
+            `helpful-links/${action}/${link.id}`,
             {}
         );
-        toaster(
-            response.message,
-            response.success ? ToastState.success : ToastState.error
-        );
-        if (mutate) void mutate();
-    };
-
-    async function toggleLinkFavorite(e: React.MouseEvent) {
-        if (!mutate) {
-            toaster('Error updating favorite status', ToastState.error);
-            return;
+        if (response.success) {
+            toaster(
+                'Helpful link state updated successfully',
```

Comment:
Can we make it so that the message doesn't use the word "state" and is clearer as to what happened?
- Helpful link favorited
- Helpful link unfavorited
- Helpful link is now hidden
- Helpful link is now visible